src/poliastro/twobody/perturbations.py

```diff
@@ -99,3 +99,65 @@ def third_body(t0, state, k, k_third, third_body):
     body_r = third_body(t0)
     delta_r = body_r - state[:3]
     return k_third * delta_r / norm(delta_r) ** 3 - k_third * body_r / norm(body_r) ** 3
+
+
+def shadow_function(r_sat, r_sun, R_Earth):
+    """Determines whether the satellite is in Earth's shadow,
+       uses algorithm 12.3 from Howard Curtis
+
+    Parameters
+    ----------
+    r_sat : numpy.ndarray
+        position of the satellite in the Earth frame (km)
```

Comment:
Same as above: we should avoid the `Earth/Sun` terminology and use `attractor/star` or similar. This should work for satellites around Mars, or around planets that orbit stars other than the Sun.